In [ ]:
#1
#escribir un arreglo del 0 al 9
import numpy as np
x=np.arange(10)

In [ ]:
x

array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

In [ ]:
#2
#segmentación para mostrar valores de 4 a 7
x[4:8]

array([4, 5, 6, 7])

In [ ]:
#3
np.split(x,[3,6])
#Separa los números del arreglo en grupos

[array([0, 1, 2]), array([3, 4, 5]), array([6, 7, 8, 9])]

####Separa los datos del arreglo en 3 grupos de datos, el primero empezando en donde empieza el arreglo, es decir 0, el segundo empezando en el 3 y el tercero empezando en el 6

In [ ]:
#4
#escribir lo siguiente en forma de data frame
import pandas as pd
data = pd.DataFrame({'Country':    ['Russia','Colombia','Chile','Equador','Nigeria'],'Rank':[121,40,100,130,11]})

In [ ]:
data

Country  Rank
0    Russia   121
1  Colombia    40
2     Chile   100
3   Equador   130
4   Nigeria    11

In [ ]:
#5
data.info()
#te da la información de cuántas columnas hay, cuántas entradas tiene el data frame, el tipo de datos que hay y la memoria que ocupa el data frame

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Country  5 non-null      object
 1   Rank     5 non-null      int64 
dtypes: int64(1), object(1)
memory usage: 208.0+ bytes


In [ ]:
#6
data = pd.DataFrame({'k1':['one']*3 + ['two']*4, 'k2':[3,2,1,3,3,4,4]})
#6a: la operación hace que se escriba la palabra "one" tres veces y una vez que terminaron las tres veces, hace que se escriba la palabra "two" cuatro veces.

In [ ]:
#6b: muestra la base de datos creada arriba
data

k1  k2
0  one   3
1  one   2
2  one   1
3  two   3
4  two   3
5  two   4
6  two   4

In [ ]:
#7
data.drop_duplicates()

k1  k2
0  one   3
1  one   2
2  one   1
3  two   3
5  two   4

##7
###Lo que realiza la operación .drop_duplicates() es que te quita los valores de tu data frame en los que se repite tanto la columna 1 como la 2, en este caso k1 y k2. Cuando el valor de k1 y k2 en el renglón 3 es igual al valor de k1 y k2 en el renglón 4, quita el renglón 4 por completo.

In [ ]:
#8
data = pd.DataFrame(np.arange(12).reshape((3, 4)),index=['CDMX', 'Guanajuato', 'Puebla'],columns=['uno', 'dos', 'tres', 'cuatro'])
data

uno  dos  tres  cuatro
CDMX          0    1     2       3
Guanajuato    4    5     6       7
Puebla        8    9    10      11

##8
###El data frame está constituido de un arreglo que va del 0 al 11 gracias a la función np.arange(12). Al agregarle el .reshape, le estamos pidiendo que acomode esos datos en nuevos valores de x,y; en este caso, 3 filas y 4 columnas. La parte del index y la parte de columns son para indicar qúe títulos queremos que lleven las filas y las columnas del data frame


In [ ]:
#9
#cambiar las etiquetas de los renglones del data frame de arriba
data = pd.DataFrame(np.arange(12).reshape((3, 4)),index=['Monterrey', 'Yucatán', 'Sonora'],columns=['uno', 'dos', 'tres', 'cuatro'])
data

uno  dos  tres  cuatro
Monterrey    0    1     2       3
Yucatán      4    5     6       7
Sonora       8    9    10      11

In [ ]:
#10
df = pd.DataFrame({'key1' : ['a', 'a', 'b', 'b', 'a'], 'key2' : ['one', 'two', 'one', 'two', 'one'], 'data1' : np.random.randn(5),'data2' : np.random.randn(5)})
df

key1 key2     data1     data2
0    a  one -0.561060  0.994252
1    a  two  0.760098 -0.301026
2    b  one -0.053361  0.672138
3    b  two  1.041775  0.575727
4    a  one -0.638772  0.228363

##10
###El data frame muestra los valores dados en las columnas que llamamos key1 y en key2. En la parte del data1 y data2, lo que se pide es que tomen 5 números aleatorios para llenar las filas del data frame.


In [ ]:
#10a
#Quita la columna de nombre "key2" del data frame
d=df.drop(columns="key2")
d

key1     data1     data2
0    a -0.561060  0.994252
1    a  0.760098 -0.301026
2    b -0.053361  0.672138
3    b  1.041775  0.575727
4    a -0.638772  0.228363

In [ ]:
#10b
#renombra los índices de el siguiente data frame
d = pd.DataFrame({'key1' : ['a', 'a', 'b', 'b', 'a'], 'data1' : np.random.randn(5),'data2' : np.random.randn(5)},
                  index=["Dato 1","Dato 2","Dato 3","Dato 4","Dato 5"])
d

key1     data1     data2
Dato 1    a -1.187340 -0.465213
Dato 2    a  0.500224  1.090630
Dato 3    b  0.440574  2.013090
Dato 4    b  1.316833  1.421605
Dato 5    a -1.285931 -0.438675

In [ ]:
#10c
#sacamos la estadística del data frame
d.describe()

data1     data2
count  5.000000  5.000000
mean   0.109736  0.433891
std    0.762971  0.493261
min   -0.638772 -0.301026
25%   -0.561060  0.228363
50%   -0.053361  0.575727
75%    0.760098  0.672138
max    1.041775  0.994252

##10c
####La media de la columna data1 es 0.109736 con una desviación estándar de 0.762971. La media de la columna data2 es 0.433891 con una desviación estándar de 0.493261. El valor máximo de data1 es 1.041775 mientras que el mínimo es -0.638772. El valor máximo de data2 es 0.994252 mientras que el mínimo es 0.301026. Los datos se pueden mover si se vuelve a correr todo porque son números aleatorios los que se generan en el data frame

In [ ]:
#10d
#calculas la media de las columnas del data frame resultante después de quitarle la columna key2
d.mean()

<ipython-input-117-8d8026cd8d72>:3: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  d.mean()


data1    0.109736
data2    0.433891
dtype: float64